timeseries/src/autogluon/timeseries/evaluator.py

```diff
@@ -16,12 +16,21 @@
 logger = logging.getLogger(__name__)
 
 
+def get_seasonal_diffs(*, y_past: pd.Series, seasonal_period: int = 1) -> pd.Series:
+    return y_past.groupby(level=ITEMID, sort=False).diff(seasonal_period).abs()
+
+
 def in_sample_seasonal_naive_error(*, y_past: pd.Series, seasonal_period: int = 1) -> pd.Series:
     """Compute seasonal naive forecast error (predict value from seasonal_period steps ago) for each time series."""
-    seasonal_diffs = y_past.groupby(level=ITEMID, sort=False).diff(seasonal_period).abs()
+    seasonal_diffs = get_seasonal_diffs(y_past=y_past, seasonal_period=seasonal_period)
     return seasonal_diffs.groupby(level=ITEMID, sort=False).mean().fillna(1.0)
 
 
+def in_sample_random_walk_error(*, y_past: pd.Series) -> pd.Series:
+    onestep_diffs = get_seasonal_diffs(y_past=y_past, seasonal_period=1)
+    return onestep_diffs.dropna().pow(2.0).groupby(level=ITEMID, sort=False).mean()
```

Comment:
1. I think we should use the seasonal error for both MASE and RMSSE (https://otexts.com/fpp3/accuracy.html#scaled-errors), i.e., not only `seasonal_period=1` for RMSSE.
2. Should we call the methods something like `in_sample_abs_seasonal_error` and `in_sample_squared_seasonal_error` to differentiate between the two cases?

Reply:
Agreed and updated.

Reply:
Nit: Should we call the variable `seasonal_diffs` now?

Reply:
Done!